### PR TITLE
feat(plugin-essentials): support multiple patterns for unlink command

### PR DESCRIPTION
**Pull request by noahnu**
**What's the problem this PR addresses?**

Addresses https://github.com/yarnpkg/berry/pull/2588#discussion_r638209031

**How did you fix it?**

Replaced leadingArgument with leadingArguments (Rest)

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/acceptance-tests/pkg-tests-specs/sources/commands/unlink.test.js [hunk 1]
```diff
@@ -227,5 +227,58 @@ describe(`Commands`, () => {
...
+          await writeJson(`${path}/packages/workspace/package.json`, {
+            name: `workspace`,
+          });
+
+          const packages = [`my-package-a`, `my-package-b`, `my-package-c`, `my-package-d`];
+          for (const pkg of packages) {
+            await writeJson(`${tmp}/${pkg}/package.json`, {
+              name: pkg,
+            });
+            await run(`link`, `${tmp}/${pkg}`, {
```
Comment (noahnu) [inline packages/acceptance-tests/pkg-tests-specs/sources/commands/unlink.test.js:250]: The link command can be extended to support multiple arguments as well

File: packages/plugin-essentials/sources/commands/unlink.ts [hunk 2]
```diff
@@ -49,41 +47,44 @@ export default class UnlinkCommand extends BaseCommand {
...
-        if (this.all) {
-          for (const workspace of project2.workspaces)
-            if (workspace.manifest.name)
-              workspacesToUnlink.add(structUtils.stringifyIdent(workspace.locator));
-
-          if (workspacesToUnlink.size === 0) {
-            throw new UsageError(`No workspace found to be unlinked in the target project`);
+    if (this.leadingArguments.length) {
+      for (const leadingArgument of this.leadingArguments) {
+        console.log(`Leading Argument`, leadingArgument);
```
Comment (paul-soporan) [inline packages/plugin-essentials/sources/commands/unlink.ts:60]:
```suggestion
```

Comment (noahnu) [reply packages/plugin-essentials/sources/commands/unlink.ts:60]: oops

File: packages/plugin-essentials/sources/commands/unlink.ts [hunk 3]
```diff
@@ -49,41 +47,43 @@ export default class UnlinkCommand extends BaseCommand {
     const topLevelWorkspace = project.topLevelWorkspace;
     const workspacesToUnlink = new Set<string>();
 
-    if (!this.leadingArgument && this.all) {
+    if (!this.leadingArguments.length && this.all) {
```
Comment (paul-soporan) [inline packages/plugin-essentials/sources/commands/unlink.ts:50]: We tend to use explicit length checks in the codebase:

```suggestion
    if (this.leadingArguments.length === 0 && this.all) {
```

File: packages/plugin-essentials/sources/commands/unlink.ts [hunk 4]
```diff
@@ -49,41 +47,43 @@ export default class UnlinkCommand extends BaseCommand {
...
-          throw new WorkspaceRequiredError(project2.cwd, absoluteDestination);
-
-        if (this.all) {
-          for (const workspace of project2.workspaces)
-            if (workspace.manifest.name)
-              workspacesToUnlink.add(structUtils.stringifyIdent(workspace.locator));
-
-          if (workspacesToUnlink.size === 0) {
-            throw new UsageError(`No workspace found to be unlinked in the target project`);
+    if (this.leadingArguments.length) {
```
Comment (paul-soporan) [inline packages/plugin-essentials/sources/commands/unlink.ts:58]:
```suggestion
    if (this.leadingArguments.length > 0) {
```

File: packages/plugin-essentials/sources/commands/unlink.ts (hunk 3 above)
Comment (noahnu) [reply packages/plugin-essentials/sources/commands/unlink.ts]: Is it worth adding this as a lint rule?

Comment (paul-soporan) [reply packages/plugin-essentials/sources/commands/unlink.ts]: It is, I think typescript-eslint has a rule requiring strict boolean expressions or something similar that would cover this case.

**Review by arcanis — APPROVED** [review]

**Pull request merged**